src/utilities/codemirror/vml/vml.test.ts

```diff
@@ -0,0 +1,568 @@
+    expect(blockNameNode).toBeDefined();
+    const block0Name = nodeContents(input, blockNameNode!);
+    expect(block0Name).toEqual('block1');
+  });
+
+  it('block library parsing', () => {
+    const input = `MODULE
+    BLOCK block1
+        INPUT arg1
+    BODY
+    END_BODY
+
+    BLOCK block2
+        INPUT STRING arg1 ; arg1 if block library includes parameter descriptions as comments
+        INPUT INT arg2 := 32 VALUES 0..127 ; arg2 if block library includes parameter descriptions as comments
+    BODY
+    END_BODY
+
+    END_MODULE`;
+
+    const commandDictionary = vmlBlockLibraryToCommandDictionary(input, 'id', '/test');
+    expect(commandDictionary.fswCommands.length).toBe(2);
+
+    expect(commandDictionary.fswCommands[0].stem).toBe('block1');
+
+    expect(commandDictionary.fswCommands[1].stem).toBe('block2');
+    expect(commandDictionary.fswCommands[1].arguments[0].description).toBe(
+      '[INPUT] arg1 if block library includes parameter descriptions as comments',
+    );
+
+    const cmd2arg2 = commandDictionary.fswCommands[1].arguments[1] as FswCommandArgumentInteger;
+    expect(cmd2arg2.name).toBe('arg2');
+    expect(cmd2arg2.default_value).toBe(32);
+    expect(cmd2arg2.range?.min).toBe(0);
+    expect(cmd2arg2.range?.max).toBe(127);
+  });
+
+  it('call', () => {
+    const input = `
+MODULE
+ABSOLUTE_SEQUENCE master_4
+FLAGS AUTOEXECUTE AUTOUNLOAD
+BODY
+A2010-020T01:23:14.0 VM_LOAD 5, "d:/seq/observe_day_39.mod"
+R10.0 SPAWN 5 observe_day_39
+A2010-020T07:34:21.0 PAUSE 5
+A2010-020T07:34:22.0 CALL dsn_contact_start "main", 1024
+A2010-020T08:11:15.0 CALL dsn_contact_end
+A2010-020T08:11:16.0 RESUME 5
+
+A2010-050T10:11:00.0 HALT 5 ; stop camera observations
+A2010-050T10:11:00.0 ISSUE VM_UNLOAD 5
+A2010-050T10:11:01.0 ISSUE VM_LOAD 1, "d:/seq/master_5.abs"
+END_BODY
+END_MODULE
+`;
+    if (allowedInVmlVersion('2.1')) {
+      assertNoErrorNodes(input, true);
+    }
+  });
+
+  it('no time tags', () => {
+    const input = `
+MODULE
+BLOCK convert_bit_rate
+INPUT bit_rate
+DECLARE INT bit_rate_index := 0
+BODY
+IF bit_rate < 6 THEN
+bit_rate_index := 0
+ELSE_IF 6 <= bit_rate && bit_rate <= 1024 THEN
+bit_rate_index := 1 + (bit_rate - 1) / 10
+ELSE
+bit_rate_index := 103
+END_IF
+RETURN bit_rate_index
+END_BODY
+BLOCK dsn_contact_start
+INPUT STRING mode VALUES "standby", "half", "full"
+INPUT bit_rate
+DECLARE INT bit_rate_index := 0
+BODY
+bit_rate_index := CALL convert_bit_rate bit_rate
+ISSUE sspa_mode standby
+R0.2 ISSUE set_power_switch sspa, on
+R1.0 ISSUE_DYNAMIC "sspa_mode", mode
+DELAY_BY gv_sspa_on_warmup_time
+R4.0 ISSUE_DYNAMIC "transponder_on", bit_rate_index
+gv_dsn_contact := TRUE
+END_BODY
+BLOCK dsn_contact_end
+BODY
+ISSUE sspa_mode standby
+ISSUE set_power_switch sspa, off
+gv_dsn_contact := FALSE
+END_BODY
+END_MODULE
+`;
+
+    if (allowedInVmlVersion('2.1')) {
+      assertNoErrorNodes(input, true);
+    }
+  });
+
+  describe('vml command info mapper', () => {
```

Comment:
I was thinking that these tests should go in its own dedicated file next to the class file. In terms of coverage, I think it's good, but I think keeping it in a file dedicated to what it's testing will help isolate any errors or regressions. The tests for `vmlTreeUtils.ts` would also be easier to find just by looking for a file next to it with an name almost exactly the same `vmlTreeUtils.test.ts`

Reply:
Addressed in https://github.com/NASA-AMMOS/aerie-ui/pull/1472/commits/90fe5669d4d59a60d8227eaaed42b9e087499146